app/models.py

```diff
@@ -96,3 +98,84 @@ def facet_plural(facet: str):
         facet_plural = facet
 
     return facet_plural
+
+
+# --------------------------------------------
+# Response model class for the knowledge panels
+# --------------------------------------------
+
+
+class ElementsForDataQualityAndLastEdits(BaseModel):
+    element_type: str
+    text_element: Optional[str] = None
+
+
+class ElementsForHungerGame(BaseModel):
+    id: int
+    element_type: str
+    text_element: str
+
+
+class ItemsForDataQualityAndLastEdits(BaseModel):
+    title: str
+    subtitle: Optional[str] = None
+    source_url: Optional[str] = None
+    elements: Optional[list[ElementsForDataQualityAndLastEdits]] = None
+
+
+class TextItemWikiData(BaseModel):
+    element_type: str
+    text_element: str
+
+
+class LinksItemWikiData(BaseModel):
+    element_type: str
+    wikipedia: Optional[str] = None
+    image_url: Optional[str] = None
+    open_street_map: Optional[str] = None
+    INAO: Optional[str] = None
+
+
+WikidataPanel = Union[TextItemWikiData, LinksItemWikiData]
+
+
+class ElementsWikidata(BaseModel):
+    id: int
+    subtitle: Optional[str] = None
+    source_url: Optional[str] = None
+    elements: Optional[list[WikidataPanel]] = None
+
+
+class WikidataKnowledgePanelItem(BaseModel):
+    title: str
+    elements: Optional[list[ElementsWikidata]] = None
+
+
+class HungerGameKnowledgePanelItem(BaseModel):
+    title: str
+    elements: Optional[list[ElementsForHungerGame]] = None
```

Comment:
There also we have a reccuring "title" "elements" attributes that should be factored in a "BasePanel" class.

Reply:
Just question how I goona put "HungerGameElement" and "WikidtaElement" in the proposed "BasePanel"

Reply:
as you make them inherit TextElement you can just tell it's a list of TextElement.

But you may also redefine "elements" in children if you want :-)